course7/homeworks/waleed/complex/complex.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <iostream>
+
+class Complex
+{
+public:
+    Complex() :
+        mR(0.0),
+        mI(0.0) {}
+	Complex(double r, double i) :
+        mR(r),
+        mI(i) {}
+    Complex operator+(const Complex& arg);
+    Complex operator-(const Complex& arg);
+    Complex operator/(const Complex& arg);
+    Complex operator*(const Complex& arg);
+
+    double GetR() const;
+
+    double GetI() const;
+
+private:
+    double mR;
+    double mI;
+};
+
+inline std::ostream& operator<<(std::ostream& os, const Complex& arg)
```

Comment:
It should be the other way around. The complex and slow operator<< should go to the C++ file, while the small, most benefiting from inlining operators should be here.